lib/curl_memory.h

```diff
@@ -138,7 +138,7 @@ extern curl_calloc_callback Curl_ccalloc;
 extern curl_wcsdup_callback Curl_cwcsdup;
 #endif
 
-#ifndef CURLDEBUG
+#if !defined(CURLDEBUG) && defined(BUILDING_LIBCURL)
```

Comment:
AFAICS the existing solution to solve this within `src` was this (e.g. in `lib/base64.c`):
```c
#ifdef BUILDING_LIBCURL
#include "curl_memory.h"
#endif
```

With this update, all those guards become redundant.
Would it make sense to delete them?

Or, stay with the existing method in this PR and update
`lib/curl_threads.c` with the same guard. Then replace
them all with this neater solution in a separate PR?